examples/00-mapdl-examples/composite_dcb.py

```diff
@@ -0,0 +1,400 @@
+mapdl.keyopt(3, 7, 0)
+mapdl.keyopt(3, 8, 0)
+mapdl.keyopt(3, 9, 0)
+mapdl.keyopt(3, 10, 0)
+mapdl.keyopt(3, 11, 0)
+mapdl.keyopt(3, 12, 3)
+mapdl.keyopt(3, 14, 0)
+mapdl.keyopt(3, 18, 0)
+mapdl.keyopt(3, 2, 0)
+mapdl.keyopt(2, 5, 0)
+
+# generating TARGE170 elements on top of cm_1
+mapdl.nsel("s", "", "", "cm_1")
+mapdl.cm("_target", "node")
+mapdl.type(2)
+mapdl.esln("s", 0)
+mapdl.esurf()
+
+# generating the CONTA174 elements on top of cm_2
+mapdl.cmsel("s", "_elemcm")
+mapdl.nsel("s", "", "", "cm_2")
+mapdl.cm("_contact", "node")
+mapdl.type(3)
+mapdl.esln("s", 0)
+mapdl.esurf()
+
+###############################################################################
+# Generating boundary conditions
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# In this final step the boundary conditions are assigned to replicate the real
+# test conditions. In particular, one end of the two composite plates
+# is fixed against translation along x, y and z. On the other hand, on the
+# opposite side of the plate the displacement conditions are applied to
+# simulate the interfacial crack opening. These conditions are applied to the
+# top and bottom nodes corresponding to the geometrical edges located at
+# x = 0.0, z = 0.0 and x = 0.0, z = 3.4 respectively. Two different components
+# are assigned to these sets of nodes for a faster identification of the nodes
+# bearing reaction forces.
+
+# applying the two displacement conditions
+mapdl.allsel()
+mapdl.nsel(type_="s", item="loc", comp="x", vmin=0.0, vmax=0.0)
+mapdl.nsel(type_="r", item="loc", comp="z", vmin=3.4, vmax=3.4)
+mapdl.d(node="all", lab="uz", value=10)
+mapdl.cm("top_nod", "node")
+
+mapdl.allsel()
+mapdl.nsel(type_="s", item="loc", comp="x", vmin=0.0, vmax=0.0)
+mapdl.nsel(type_="r", item="loc", comp="z", vmin=0.0, vmax=0.0)
+mapdl.d(node="all", lab="uz", value=-10)
+mapdl.cm("bot_nod", "node")
+
+# applying the fix condition
+mapdl.allsel()
+mapdl.nsel(type_="s", item="loc", comp="x", vmin=85.0, vmax=85.0)
+mapdl.d(node="all", lab="ux", value=0.0)
+mapdl.d(node="all", lab="uy", value=0.0)
+mapdl.d(node="all", lab="uz", value=0.0)
+
+mapdl.eplot(
+    plot_bc=True, bc_glyph_size=3, title="", background="white", show_axes=False
+)
+
+###############################################################################
+# Solve the non-linear static analysis
+# ====================================
+#
+# A non-linear static analysis is run to in this very last step. One hundreds
+# substeps are requested in this case to have a smooth crack opening
+# progression as well as to facilitate convergency for the static solver.
+
+# entering the solution processor and defining the analysis settings
+mapdl.allsel()
+mapdl.finish()
+mapdl.run("/SOLU")
+mapdl.antype("static")
+
+# activating non-linear geometry
+mapdl.nlgeom("on")
+
+# requesting substeps
+mapdl.autots(key="on")
+mapdl.nsubst(nsbstp=100, nsbmx=100, nsbmn=100)
+mapdl.kbc(key=0)
+mapdl.outres("all", "all")
+
+# solving
+output = mapdl.solve()
+print(output)
```

Comment:
This is going to print a lot of:
```text
 *** WARNING ***                         CP =       0.000   TIME= 00:00:00
 Coupled node set 1 has unused node 35.
```
also:
```text
 *** WARNING ***                         CP =       0.000   TIME= 00:00:00
 Specified degree of freedom constraint UY at unused node 27.
```
It is not big deal, but I wonder if we can avoid them or at least, some of them.

Reply:
removed